src/software/ai/hl/stp/tactic/goalie/goalie_tactic_test.py

```diff
@@ -171,5 +171,79 @@ def test_goalie_blocks_shot(
     )
 
 
+@pytest.mark.parametrize(
+    "ball_position",
+    [
+        (tbots_cpp.Point(-3.4, 0)),
+        (tbots_cpp.Point(-3.4, 0.9)),
+        (tbots_cpp.Point(-4.0, 1.1)),
+    ],
+)
+def test_goalie_clears_from_dead_zone(
+    ball_position, simulated_test_runner,
+):
+    # Setup Robot
+    simulated_test_runner.simulator_proto_unix_io.send_proto(
+        WorldState,
+        create_world_state(
+            [],
+            blue_robot_locations=[
+                tbots_cpp.Field.createSSLDivisionBField().friendlyDefenseArea().centre()
+            ],
+            ball_location=ball_position,
+            ball_velocity=tbots_cpp.Vector(0, 0),
+        ),
+    )
+
+    # Setup Tactic
+    params = AssignedTacticPlayControlParams()
+    params.assigned_tactics[0].goalie.CopyFrom(
+        GoalieTactic(max_allowed_speed_mode=MaxAllowedSpeedMode.PHYSICAL_LIMIT)
+    )
+    simulated_test_runner.blue_full_system_proto_unix_io.send_proto(
+        AssignedTacticPlayControlParams, params
+    )
+
+    # Setup no tactics on the enemy side
+    params = AssignedTacticPlayControlParams()
+    simulated_test_runner.yellow_full_system_proto_unix_io.send_proto(
+        AssignedTacticPlayControlParams, params
+    )
+
+    # Always Validation
+    always_validation_sequence_set = [
+        [
+            # RobotAlwaysStaysInRegion(
+            #     regions=[tbots_cpp.Field.createSSLDivisionBField().friendlyDefenseArea()]
+            # ),
+        ]
+    ]
+
+    # Eventually Validation
+    eventually_validation_sequence_set = [
+        [
+            BallEventuallyEntersRegion(
+                regions=[
+                    tbots_cpp.Field.createSSLDivisionBField().friendlyDefenseArea()
+                ]
```

Comment:
I would suggest replacing this `BallEventuallyEntersRegion` with checking if the ball gets moved (ball velocity > 0)

Reply:
actually we don't really need a BallEventuallyEntersRegion cost function, BallEventuallyExitsRegion should be sufficient